core/base/src/main/java/org/eclipse/dataspaceconnector/core/health/HealthCheckServiceImpl.java

```diff
@@ -28,7 +29,8 @@
     private final ScheduledExecutorService executor;
     private final HealthCheckServiceConfiguration configuration;
 
-    public HealthCheckServiceImpl(HealthCheckServiceConfiguration configuration) {
+    public HealthCheckServiceImpl(HealthCheckServiceConfiguration configuration,
```

Comment:
Isn't it good to encapsulate the details of requiring a `ScheduledExecutorService` within the class itself? I can do this change, but my impression is would be leaking too many implementation details.

Reply:
Generally I agree, but here we're not only leaking the usage of instrumentation, we're also forcing users of this class to pass in a dummy implementation if no instr. is needed.

Reply:
Yes I think it is in the nature of instrumentation to "leak".
Perhaps we could have a utility value object to pass around the usual suspects: instrumentation, monitor etc.

Reply:
I'm not too concerned about future refactorings, but it feels a bit clumsy to have to explicitly pass an object that fulfills a cross-cutting (and unrelated) concern into a constructor, especially since it's the only occurrence and most of the other instrumentations happen transparently, and we have such an elegant solution as decorating the executor.

Reply:
I would suggest leaving it as is for now. We can revisit it once we add instrumentation for further executors, which I would anyway do rather sooner than later for the executors in state machines.